gaia_tools/load/spec.py

```diff
@@ -0,0 +1,143 @@
+        int(file[file.find("_") + 1 : file.rfind("-")]) for file in file_names
+    ]
+    healpix_8_max = [
+        int(file[file.rfind("-") + 1 : file.rfind(".csv")]) for file in file_names
+    ]
+    reference_file = {
+        "file": file_names,
+        "healpix8_min": healpix_8_min,
+        "healpix8_max": healpix_8_max,
+    }
+
+    combo_comparisons = np.asarray(
+        [
+            (reference_file["healpix8_min"] <= i)
+            & (i <= reference_file["healpix8_max"])
+            for i in reduced_source_ids
+        ]
+    )
+    files_required = np.unique(
+        [list(compress(file_names, i)) for i in combo_comparisons]
+    )
+
+    if not np.all(np.any(combo_comparisons, axis=1)):
+        raise ValueError("Contain invalid Gaia source id")
+
+    for idx, i in enumerate(tqdm.tqdm(file_names, desc="Working on data file:")):
+        if not i in files_required:
+            continue
+        else:
+            spec_f = ascii.read(os.path.join(base_path, i))
+            current_source_ids = source_ids[combo_comparisons[:, idx]]
+            matches, idx1, idx2 = np.intersect1d(
+                current_source_ids,
+                spec_f["source_id"].data,
+                assume_unique=False,
+                return_indices=True,
+            )
+            if len(matches) > 0:
+                current_idx = np.where(combo_comparisons[:, idx])[0][idx1]
+                all_spec[current_idx] = np.vstack(spec_f["flux"][idx2])
+                all_spec_error[current_idx] = np.vstack(spec_f["flux_error"][idx2])
+                not_found[current_idx] = False
+
+    # deal with duplicated source_id
+    if not assume_unique:
+        uniques = np.unique(source_ids, return_index=True)[1]
+        result = np.ones_like(source_ids, dtype=bool)
+        result[uniques] = False
+        duplicated_idx = np.where(result)[0]
+        for i in duplicated_idx:
+            first_occurence_idx = np.argmax(source_ids == source_ids[i])
+            all_spec[i] = all_spec[first_occurence_idx]
+            all_spec_error[i] = all_spec_error[first_occurence_idx]
+            not_found[i] = not_found[first_occurence_idx]
+            
+    if np.any(not_found):
+        warnings.warn(f"These source id have no corresponding spectra found: {source_ids[not_found]}")
+
+    return wavelength_grid, all_spec, all_spec_error
+
+
+def load_rvs_spec(source_ids, assume_unique=False):
+    """
+    NAME:
+        load_rvs_spec
+    PURPOSE:
+        Read corresponding RVS spectra for a list of source id
+    INPUT:
+        source_ids (int, list, ndarray): source id
+        assume_unique (bool): whether to assume the list of source id is unique
+    OUTPUT:
+        wavelength grid, RVS spectra flux row matched to source_id, RVS spectra corresponding flux uncertainty
+    HISTORY:
+        2022-06-16 - Written - Henry Leung (UofT)
+    """
+    base_path = os.path.join(
+        _GAIA_TOOLS_DATA, "Gaia/gdr3/Spectroscopy/rvs_mean_spectrum/"
+    )
```

Comment:
Does this work on Windows? Normally, one has to do
```
os.path.join(
        _GAIA_TOOLS_DATA, "Gaia","gdr3","Spectroscopy","rvs_mean_spectrum"
    )
```
to get the slashes right, but maybe that's not actually necessary?

Reply:
It works on windows but I have switched to your suggestion just to be safe.